EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -135,63 +142,146 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
-        /// </summary>
-        /// <param name="psms"></param>
-        /// <returns></returns>
-        public static Dictionary<string, int> GetAccessionCounts(List<PeptideSpectralMatch> psms)
-        {
-            Dictionary<string, int> accessionCountDictionary = new Dictionary<string, int>();
-
-            foreach (PeptideSpectralMatch psm in psms)
-            {
-                if (psm != null)
+                foreach (PeptideSpectralMatch psm in psms.Where(f => (f.FullFilePath == filename || f.FullFilePath == null) && f.FdrInfo.QValue <= 0.01))
                 {
-                    foreach (var (Notch, Peptide) in psm.BestMatchingPeptides)
+                    foreach ((int notch, PeptideWithSetModifications pwsm) in psm.BestMatchingPeptides)
                     {
-                        string accession = Peptide.Protein.Accession;
-                        if (accession != null)
+                        if (pwsm.BaseSequence.Equals(pwsm.FullSequence) && !modified)
+                        {
+                            double predictedHydrophobicity = calc.ScoreSequence(pwsm);
+                            int possibleKey = (int)Math.Round(psm.ScanRetentionTime, 0);
+                            if (hydrobophobicites.ContainsKey(possibleKey))
+                            {
+                                hydrobophobicites[possibleKey].Add(predictedHydrophobicity);
+                            }
+                            else
+                            {
+                                hydrobophobicites.Add(possibleKey, new List<double>() { predictedHydrophobicity });
+                            }
+                        }
+                        else if (!pwsm.BaseSequence.Equals(pwsm.FullSequence) && modified)
                         {
-                            if (accessionCountDictionary.Keys.Contains(accession))
+                            double predictedHydrophobicity = calc.ScoreSequence(pwsm);
+                            int possibleKey = (int)Math.Round(psm.ScanRetentionTime, 0);
+                            if (hydrobophobicites.ContainsKey(possibleKey))
                             {
-                                accessionCountDictionary[accession]++;
+                                hydrobophobicites[possibleKey].Add(predictedHydrophobicity);
                             }
                             else
                             {
-                                accessionCountDictionary.Add(accession, 1);
+                                hydrobophobicites.Add(possibleKey, new List<double>() { predictedHydrophobicity });
                             }
                         }
                     }
                 }
+
+                foreach (int key in hydrobophobicites.Keys)
+                {
+                    averagesCommaStandardDeviations.Add(key, new Tuple<double, double>(hydrobophobicites[key].Average(), hydrobophobicites[key].StandardDeviation()));
```

Comment:
interquartile range is more robust to outliers than std dev but this doesn't need to be optimized now